### DISCUSSION TITLE
How does sioyek concatenate lines accurately when copying multi-line text?

### DISCUSSION THREAD

**Discussion by felixnie** [ahrm/sioyek#665] [General]
I've always found copying multi-line text annoying, as most PDF readers can only parse the lines as seperate strings with End of Line in between. I'm surprised on how sioyek deal with this issue neatly.

May I ask what's the barrier on accurate multi-line parsing/copying, and what's the solution for this problem?

I've been using Zotero/Foxit/Chrome/Edge PDF readers and none of them can achieve accurate multi-line parsing. Please let me know if I'm wrong or you know some other readers that have solutio for this.

Thanks to anyone who replies.

**Comment by ahrm** [comment]
> May I ask what's the barrier on accurate multi-line parsing/copying, and what's the solution for this problem?

Nothing. It is extremely easy to implement. I assume the main reason other PDF viewers don't do this is because the simply don't care.
